app/api/chemotion/molecule_api.rb

```diff
@@ -188,9 +188,8 @@ class MoleculeAPI < Grape::API
             end
           end
         end
+        
         molecule&.attributes&.merge(temp_svg: svg_process[:svg_file_name], ob_log: ob)
-
-        present molecule, with: Entities::MoleculeEntity
```

Comment:
Why did you remove this line? The change looks rather random... The API should present their objects using Grape Entities. Otherwise you get pretty much undefined results, missing relations defined in MoleculeIdentity etc.

Reply:
I see. But then the correct move would be to add these additional attributes to the entity (using the context hash allows this, see ApplicationEntity#displayed_in_list? for how to implement it), not removing the entity entirely.

Returning the attributes hash directly is undefined behaviour as new attributes are added or old ones deleted over time. Using the entity, the returned data structure can be inferred directly without looking up the current data model of the molecule table.

Reply:
I updated it like you said. Please help me to take a loook.

Reply:
This should work fine :) The options hash access evaluates to nil so any other place using the MoleculeEntity will always get nil. While it is possible to only render the temp_svg and ob_log (what is an ob_log anyway? could we refactor this variable here and in all other related places to have a meaningful name?) only when not nil, I don't consider that strictly necessary.

Reply:
While it is possible to only render the temp_svg and ob_log - Could you show me how to achieve it?
What is an ob_log anyway? could we refactor this variable here and in all other related places to have a meaningful name? - Actually I don't know either, I fix here for it to run correctly at first, we can improve it later.

Reply:
The MoleculeEntity actually contains an example of how it works :) see the separate exposure of  :molecule_names ? the :unless parameter receives a Proc with the options as parameter, so you could write it like this:

`expose :temp_svg, unless: ->(instance, options) { options[:temp_svg].present? }`

Reply:
updated

Reply:
👍